### PR TITLE
No Deprecation and new file structure

### LINKED ISSUE #310: New File Structure

### LINKED ISSUE #311: No Deprecation

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
fix #311 
fix #310

and add some more information in the upgrade file.
For removing the deprecations the admin dependencies of routing bundle have to be removed:

/requires https://github.com/symfony-cmf/routing-bundle/pull/369

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,16 +14,16 @@ cache:
...
+    - SYMFONY_VERSION=3.2.*
   global:
     - SYMFONY_DEPRECATIONS_HELPER=weak
 
 matrix:
   include:
     - php: 5.6
       env: DEPS=dev SYMFONY_VERSION=3.1.*
-    - php: 5.5
-      env: COMPOSER_FLAGS="--prefer-lowest"
```
Comment (dbu) [inline .travis.yml]: we have to keep one prefer-lowest test run

File: UPGRADE-2.0.md [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+ * The Admin extension to edit seo metadata was moved into `symfony-cmf/sonata-admin-integration`.
+   The change of the namespace is handled in the service definition. So you have to change the usage of the
+   service.
+   
+   Before:
+   
+   ```yaml
+        # app/config/config.yml
+        
+        sonata_admin:
```
Comment (dbu) [inline UPGRADE-2.0.md]: that looks wrong. the admin was configured under cmf_menu or cmf_core

File: composer.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
         "php": "^5.5.9|^7.0",
         "symfony/framework-bundle": "^2.8|^3.0",
         "sonata-project/seo-bundle": "^1.1.3|^2.0",
-        "doctrine/collections": "^1.0"
+        "doctrine/collections": "^1.0",
+        "symfony-cmf/routing-bundle": "dev-move_sonata_only"
```
Comment (dbu) [inline composer.json]: this has been merged, you can switch to `^2.0`

File: composer.json [hunk 4]
```diff
@@ -44,7 +44,8 @@
     },
     "autoload": {
         "psr-4": {
-            "Symfony\\Cmf\\Bundle\\SeoBundle\\": ""
+            "Symfony\\Cmf\\Bundle\\SeoBundle\\": "src/",
+            "Symfony\\Cmf\\Bundle\\SeoBundle\\Tests\\": "tests/"
```
Comment (dbu) [inline composer.json]: this should only be in autoload-dev

File: UPGRADE-2.0.md (hunk 2 above)
Comment (ElectricMaxxx) [inline UPGRADE-2.0.md]: right. Realized it as i started moving sonata from cora-bundle (80% done). i will change that.

File: composer.json (hunk 3 above)
Comment (ElectricMaxxx) [inline composer.json]: :+1:
``

File: composer.json (hunk 4 above)
Comment (ElectricMaxxx) [inline composer.json]: :+1:

File: UPGRADE-2.0.md (hunk 2 above)
Comment (ElectricMaxxx) [inline UPGRADE-2.0.md]: rollback. core-bundle and menu-bundle simply add their own information to the extension. The example i gave is the enalbling of the extension, but it does not look good at all.

**Comment by ElectricMaxxx** [comment]
got a strange test result after choosing master of routing-bundle again.

File: .travis.yml [hunk 5]
```diff
@@ -14,7 +14,9 @@ cache:
 
 env:
   matrix:
-    - SYMFONY_VERSION=2.8.* 
+    - SYMFONY_VERSION=2.8.*
+    - SYMFONY_VERSION=3.1.*
```
Comment (dbu) [inline .travis.yml]: please don't do this but instead add to the matrix.include below. if you add symfony versions up here, the tests are ran for each php version X every symfony version, which leads to a ton of unnecessary builds. i guess we should build the latest symfony (aka 3.2.*) with every php version, and add the others below to be built once

File: UPGRADE-2.0.md [hunk 6]
```diff
@@ -1,6 +1,36 @@
 UPGRADE FROM 1.x to 2.0
 =======================
 
+### SonataAdmin Support
+
+ * The Admin extension to edit seo metadata was moved into `symfony-cmf/sonata-admin-integration`.
```
Comment (dbu) [inline UPGRADE-2.0.md]: sonata-admin-integration-bundle

File: UPGRADE-2.0.md [hunk 7]
```diff
@@ -1,6 +1,36 @@
...
+   The change of the namespace is handled in the service definition. So you have to change the usage of the
```
Comment (dbu) [inline UPGRADE-2.0.md]: maybe say "With the move, the admin extension service names also changed. If you are using the `cmf_seo.admin_extension` service, you need to adjust your configuration."

File: composer.json [hunk 8]
```diff
@@ -40,11 +40,17 @@
         "doctrine/phpcr-bundle": "To persist the metadata in PHPCR ODM documents",
         "doctrine/doctrine-bundle": "To persist the metadata in ORM entities",
         "doctrine/orm": "To persist the metadata in ORM entities, ~2.4",
-        "burgov/key-value-form-bundle": "To edit the seo metadata 'extra' fields"
+        "burgov/key-value-form-bundle": "To edit the seo metadata 'extra' fields",
+        "symfony-cmf/sonata-admin-integration": "To provide an admin interface for the PHPCR ODM documents."
```
Comment (dbu) [inline composer.json]: -bundle

Comment (ElectricMaxxx) [inline composer.json]: i think that bundle isn't important for seo anymore. It is gone with the admin.

File: .travis.yml [hunk 9]
```diff
@@ -14,7 +15,7 @@ cache:
 
 env:
   matrix:
-    - SYMFONY_VERSION=2.8.* 
+    - SYMFONY_VERSION=3.1.*
   global:
     - SYMFONY_DEPRECATIONS_HELPER=weak
```
Comment (wouterj) [inline .travis.yml]: missing `- SYMFONY_PHPUNIT_REMOVE="symfony/yaml" SYMFONY_PHPUNIT_DIR=".phpunit"` and then also adding `.phpunit` to the cache directories.

File: .travis.yml [hunk 10]
```diff
@@ -26,6 +27,12 @@ matrix:
       env: COMPOSER_FLAGS="--prefer-lowest"
     - php: 5.6
       env: DEPS=dev COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3.0.*
+    - php: 5.6
+      env: DEPS=dev SYMFONY_VERSION=3.2.*
+    - php: 7.0
+      env: DEPS=dev SYMFONY_VERSION=3.2.*
+    - php: 7.1
+      env: DEPS=dev SYMFONY_VERSION=3.2.*
```
Comment (wouterj) [inline .travis.yml]: These should be removed. We should only test one Symfony version against all PHP version and one PHP version against all Symfony versions.

File: .travis.yml [hunk 11]
```diff
@@ -4,28 +4,33 @@ php:
...
 
 matrix:
   include:
-    - php: 5.6
-      env: DEPS=dev SYMFONY_VERSION=3.1.*
     - php: 5.5
       env: COMPOSER_FLAGS="--prefer-lowest"
     - php: 5.6
-      env: DEPS=dev COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3.0.*
+      env: DEPS=dev COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3..*
```
Comment (dbu) [inline .travis.yml]: `3..* seems wrong. did you mean 3.*?

File: .travis.yml [hunk 12]
```diff
@@ -4,28 +4,33 @@ php:
...
+    - php: 7.0
+      env: DEPS=dev SYMFONY_VERSION=3.2.*
+    - php: 7.1
+      env: DEPS=dev SYMFONY_VERSION=3.2.*
```
Comment (dbu) [inline .travis.yml]: this is not needed, php 7.1 is in the matrix above

File: .travis.yml (hunk 11 above)
Comment (ElectricMaxxx) [inline .travis.yml]: no 3.2.* :-)

File: .travis.yml (hunk 12 above)
Comment (ElectricMaxxx) [inline .travis.yml]: indeed but not with 3.2.* of symfony. You and @wouterj said:
- One Symfony Version with each php version
- One PHP Version with each Symfony version

f... you are right and no i understand your both comments
The matrix is "One Symfony Version with each PHP version", so i need to add the other one.

File: .travis.yml [hunk 13]
```diff
@@ -4,28 +4,35 @@ php:
...
 
 matrix:
   include:
-    - php: 5.6
-      env: DEPS=dev SYMFONY_VERSION=3.1.*
     - php: 5.5
       env: COMPOSER_FLAGS="--prefer-lowest"
     - php: 5.6
-      env: DEPS=dev COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3.0.*
+      env: DEPS=dev COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3.1.*
```
Comment (wouterj) [inline .travis.yml]: This should become:
```php
- php: nightly
  env: DEPS=dev SYMFONY_VERSION=3.3.*
```

File: .travis.yml [hunk 14]
```diff
@@ -4,28 +4,35 @@ php:
...
+    - php: 7.1
+      env: DEPS=dev SYMFONY_VERSION=2.8.*
+    - php: 7.1
+      env: DEPS=dev SYMFONY_VERSION=3.0.*
+    - php: 7.1
+      env: DEPS=dev SYMFONY_VERSION=3.1.*
```
Comment (wouterj) [inline .travis.yml]: please remove `DEPS=dev` from all these

**Comment by ElectricMaxxx** [comment]
@symfony-cmf/contributors no clue what happens here. That the nightly test fails sounds normal. But both others 5.5/prefer-lowes and 7.0/Symfony 2.8 seems to get a non master version of routing-bundle or something differen, where there still is `use_sonata_admin` configured somewhere.

**Comment by ElectricMaxxx** [comment]
https://github.com/phpcr/phpcr-utils/pull/165 should fix the last issue.

File: composer.json [hunk 15]
```diff
@@ -15,7 +15,8 @@
         "php": "^5.5.9|^7.0",
         "symfony/framework-bundle": "^2.8|^3.0",
         "sonata-project/seo-bundle": "^1.1.3|^2.0",
-        "doctrine/collections": "^1.0"
+        "doctrine/collections": "^1.0",
+        "phpcr/phpcr-utils": "^1.2.9"
```
Comment (ElectricMaxxx) [inline composer.json]: just wanna give that a try, will remove it later. Just wanne see a green build before i have to go to bed - hopefully one for tonight.

Comment (dbu) [inline composer.json]: php 7 now works. 7.1 keeps having the error, but says it installed 1.2.9 so i really don't know how that can happen. lets try again tomorrow. (here in vancouver, its only 16:30 :-) )

Comment (ElectricMaxxx) [inline composer.json]: maybe we should create a 2.0 for phpcr-utils by dumping php < 5.6? We can instantiate every array with [] them.

Comment (dbu) [inline composer.json]: this has nothing to do with short array notation or not. i really don't get how 7.1 could get the error we see, it makes no sense if itreally installed 1.2.9 - or is the error in some other place this time?

File: composer.json [hunk 16]
```diff
@@ -29,22 +29,26 @@
         "symfony-cmf/core-bundle": "^1.1|^2.0",
         "matthiasnoback/symfony-dependency-injection-test": "^0.7.4",
         "matthiasnoback/symfony-config-test": "^1.3.1",
+        "symfony/phpunit-bridge": "^3.2",
         "burgov/key-value-form-bundle": "^1.0"
     },
-    "minimum-stability": "dev",
```
Comment (dbu) [inline composer.json]: i think with this things are not installable anymore on travis, until we have released things

Comment (ElectricMaxxx) [inline composer.json]: just was a try to get it working at all.

**Comment by ElectricMaxxx** [comment]
As this one is completelly broken, i will divide it into:

* upgrade file
* php 5.6/travis update
* file structure

i will close/delete this one later and chery pick locally.

**Comment by ElectricMaxxx** [comment]
arg some thin missed on rebase.

**Comment by dbu** [comment]
in a pr like this, i would squash commits before rebasing... do you think you can sort this out or shall i try?

**Comment by ElectricMaxxx** [comment]
cei am away from a pc,, if you would do, would be nice

**Comment by dbu** [comment]
rebased. now we have the stupid phpcr-utils error again. i looked into it and either we had some hickup with two variants of 1.2.9 or i completly goofed this up and never released a version with the fix. tagged 1.2.10 which hopefully fixes this problem for good.

**Comment by dbu** [comment]
now we get The file "/home/travis/build/symfony-cmf/seo-bundle/tests/Resources/app/config/../../../../Resources/config/routing/sitemap.xml" does not exist in /home/travis/build/symfony-cmf/seo-bundle/tests/Resources/app/config/../../../../Resources/config/routing/sitemap.xml (which is loaded in resource "/home/travis/build/symfony-cmf/seo-bundle/tests/Resources/app/config/../../../../Resources/config/routing/sitemap.xml". are we missing a src/ here?

**Comment by ElectricMaxxx** [comment]
you need an extra `../` in the routing call. as we move it to src/ and test/  we are one folder deepeer. seems to be gone while merging/rebasing.

**Comment by ElectricMaxxx** [comment]
@dbu do you want to fix that? After the rebase the branch is on your side :-)

**Comment by ElectricMaxxx** [comment]
@dbu i did it

**Comment by ElectricMaxxx** [comment]
Had to remove the nightly build. I think the fail was in a form component and not on our side.

**Pull request merged**